library/core/src/intrinsics.rs

```diff
@@ -991,14 +995,16 @@ extern "rust-intrinsic" {
     ///
     /// let v_clone = v_orig.clone();
     ///
-    /// // The no-copy, unsafe way, still using transmute, but not relying on the data layout.
-    /// // Like the first approach, this reuses the `Vec` internals.
-    /// // Therefore, the new inner type must have the
-    /// // exact same size, *and the same alignment*, as the old type.
-    /// // The same caveats exist for this method as transmute, for
-    /// // the original inner type (`&i32`) to the converted inner type
-    /// // (`Option<&i32>`), so read the nomicon pages linked above and also
-    /// // consult the [`from_raw_parts`] documentation.
+    /// // To transmute the inner type of the contents of a container, you must make sure
+    /// // to not violate any of the container's invariants. For `Vec`, this means that
+    /// // both the size *and alignment* of the inner types match. Other containers could
+    /// // rely on different properties, even the `TypeId`, where changing the type would
+    /// // not be possible without violating the container invariants.
```

Comment:
Nice! This text sounds like it should be put out of the code, into the text just before.